Cython/Compiler/Parsing.py

```diff
@@ -3002,9 +3002,15 @@ def p_with_gil(s):
         return 0
 
 def p_exception_value_clause(s):
+    # the default behaviour is to always
+    # check for exceptions
+    exc_check = 1
     exc_val = None
-    exc_check = 0
-    if s.sy == 'except':
+    if s.sy == 'IDENT' and s.systring == 'noexcept':
```

Comment:
Not sure this is correct; before we weren't looking for `s.sy == 'IDENT'` preceding an `except`; the placement of `noexcept` is the same, why the need to also check `'IDENT'`?

Reply:
Thanks! I might be missing something, but I just borrowed from other parts of the parser that parse out other C/C++/Cython symbols. They all seem to do checks like `s.sy == 'IDENT' and s.systring == 'cppclass'`.

Reply:
Without having checked in detail, this looks right to me.

It's 'IDENT' because it isn't specifically a reserved keyword (because you should be able to use `noexcept` as a Python variable name for example). You check for that, then you check the content. If you didn't check for `IDENT` first then I don't know if you'd be confident it has a `systring` attribute.

Not completely sure of any of this but it definitely looks reasonable on a first look.

Reply:
The difference is that `except` is a Python keyword, so gets picked out as special at an earlier stage...

Reply:
That makes total sense -- thanks!